server/routes.go

```diff
@@ -87,43 +80,42 @@ func generate(c *gin.Context) {
 	}
 
 	ch := make(chan string)
-	model.SetTokenCallback(func(token string) bool {
-		ch <- token
-		return true
-	})
-
-	predictOpts := getPredictOpts(req)
-
-	go func() {
+	g, _ := errgroup.WithContext(c.Request.Context())
+	g.Go(func() error {
 		defer close(ch)
-		_, err := model.Predict(req.Prompt, predictOpts)
-		if err != nil {
-			panic(err)
-		}
-	}()
+		return llm.Predict(req.Prompt, func(s string) error {
+			ch <- s
```

Comment:
looks like this cant return an error?

Reply:
for now though it'll most likely have more error handling in the future. can remove the error return for now